src/main/java/de/medizininformatik_initiative/process/data_sharing/DataSharingProcessPluginDeploymentStateListener.java

```diff
@@ -1,48 +1,200 @@
+
 			dmsFhirClientFactory.testConnection();
 			keyProvider.createPublicKeyIfNotExists();
 		}
 	}
+
+	private <T extends MetadataResource> void updateOlderResourcesIfCurrentIsNewestResource(String url, Class<T> type,
+			BiConsumer<T, List<T>> converter)
+	{
+		Bundle searchResult = search(type, url);
+		List<T> allResources = extractResources(searchResult, type, url);
+
+		T currentResource = filterCurrentResource(allResources, type, url);
+		List<T> olderNewerResources = filterOlderNewerResources(allResources);
+
+		if (currentIsNewestResource(olderNewerResources))
+		{
+			converter.accept(currentResource, olderNewerResources);
+		}
+	}
+
+	private Bundle search(Class<? extends Resource> type, String url)
+	{
+		return api.getFhirWebserviceClientProvider().getLocalWebserviceClient().search(type,
+				Map.of("url", List.of(url)));
+	}
+
+	private <T extends MetadataResource> List<T> extractResources(Bundle bundle, Class<T> type, String url)
+	{
+		return bundle.getEntry().stream().filter(Bundle.BundleEntryComponent::hasResource)
+				.map(Bundle.BundleEntryComponent::getResource).filter(type::isInstance).map(type::cast)
+				.filter(m -> url.equals(m.getUrl())).toList();
+	}
+
+	private <T extends MetadataResource> T filterCurrentResource(List<T> all, Class<T> type, String codeSystemUrl)
+	{
+		return all.stream().filter(m -> resourcesVersion.equals(m.getVersion())).findFirst().orElseThrow(
+				() -> new RuntimeException(type.getSimpleName() + " " + codeSystemUrl + "|" + resourcesVersion));
+	}
+
+	private <T extends MetadataResource> List<T> filterOlderNewerResources(List<T> all)
+	{
+		return all.stream().filter(c -> !resourcesVersion.equals(c.getVersion())).toList();
+	}
+
+	private boolean currentIsNewestResource(List<? extends MetadataResource> olderNewerResources)
+	{
+		return olderNewerResources.stream().noneMatch(this::isNewerResource);
+	}
+
+	private <T extends MetadataResource> boolean isNewerResource(T resource)
+	{
+		MinorMajorVersion current = getMajorMinorVersion(resourcesVersion);
+		MinorMajorVersion olderNewer = getMajorMinorVersion(resource.getVersion());
+
+		return current.major <= olderNewer.major && current.minor < olderNewer.minor;
+	}
+
+	private MinorMajorVersion getMajorMinorVersion(String version)
+	{
+		if (version.matches("\\d\\.\\d"))
+		{
+			String[] minorMajor = version.split("\\.");
+			return new MinorMajorVersion(Integer.parseInt(minorMajor[0]), Integer.parseInt(minorMajor[1]));
+		}
+
+		throw new RuntimeException("Fhir resource version " + version + " does not match regex \\d\\.\\d");
+	}
+
+	private BiConsumer<CodeSystem, List<CodeSystem>> adaptCodeSystems()
+	{
+		return (currentResource, olderResources) ->
+		{
+			List<CodeSystem> codeSystemsWithNonMatchingConceptCodes = filterCodeSystemsWithNonMatchingConceptCodesAndAdaptToCurrentCodeSystemConceptCodes(
+					currentResource, olderResources);
+			updateResources(codeSystemsWithNonMatchingConceptCodes);
+		};
+	}
```

Comment:
Since the methods `adaptCodeSystems` and `adaptStructureDefinitionExtensions` don't have any parameters, I would write them as "normal" methods and use method references in `onProcessesDeployed`.
```suggestion
	private void adaptCodeSystems(CodeSystem currentResource, List<CodeSystem> olderResources)
	{
		List<CodeSystem> codeSystemsWithNonMatchingConceptCodes = filterCodeSystemsWithNonMatchingConceptCodesAndAdaptToCurrentCodeSystemConceptCodes(
					currentResource, olderResources);
		updateResources(codeSystemsWithNonMatchingConceptCodes);
	}
```

Reply:
implemented with https://github.com/medizininformatik-initiative/mii-processes-common/commit/d86c95b1c5771d2e5f5e6712d8c93734e16b5f17

Reply:
implemented with e7102a6